### PR TITLE
Upgrade Resource Detector version in Distro

### PR DESCRIPTION
**Pull request by jeremydvoss**
# Description

Upgrade Distro to new VM Resource detector. Requires OTel 1.21/0.42b0 and exporter 1.0.0b21.

Relies on https://github.com/Azure/azure-sdk-for-python/pull/33865

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+## 1.0.0b22 (Unreleased)
+
+### Features Added
+
+### Breaking Changes
+
+### Bugs Fixed
+
+### Other Changes
+
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:12]: Automatic PR failed: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3410861&view=logs&j=9e408172-77a7-548b-8cd1-e45278a8c989&t=afb6697a-66be-5ac1-3938-caa677f9c3f8&l=12

File: sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_version.py [hunk 2]
```diff
@@ -5,4 +5,4 @@
 # license information.
 # --------------------------------------------------------------------------
 
-VERSION = "1.0.0b21"
+VERSION = "1.0.0b22"
```
Comment (jeremydvoss) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_version.py:8]: Automatic PR failed: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3410861&view=logs&j=9e408172-77a7-548b-8cd1-e45278a8c989&t=afb6697a-66be-5ac1-3938-caa677f9c3f8&l=12

File: sdk/monitor/azure-monitor-opentelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,8 @@
     ([#33761](https://github.com/Azure/azure-sdk-for-python/pull/33761))
 - Fix siteName in diagnostic logging
     ([#33808](https://github.com/Azure/azure-sdk-for-python/pull/33808))
+- Update to opentelemetry-resource-detector-azure 0.1.1 and OTel SKD/API 1.21
```
Comment (lzchen) [inline sdk/monitor/azure-monitor-opentelemetry/CHANGELOG.md:17]:
```suggestion
- Update min dependency versions opentelemetry-resource-detector-azure~=0.1.1, exporter~=1.0.0b21, OTel SDK/API~= 1.21
```

File: sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py [hunk 4]
```diff
@@ -92,7 +93,7 @@ def configure_azure_monitor(**kwargs) -> None:
 
 
 def _setup_tracing(configurations: Dict[str, ConfigurationValue]):
-    resource = configurations[RESOURCE_ARG] # type: ignore
+    resource =  cast(Resource, configurations[RESOURCE_ARG]) # type: ignore
```
Comment (lzchen) [inline sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py:96]: Why do we need this cast?

Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py:96]: Mypy complained. I am not sure why it waited until now. But casting the condigurations from Union to a clear type fixes it.
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3413933&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=e3f8f308-6e5f-58aa-3a2a-fe425958de32&l=360

Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py]: Changed

File: sdk/monitor/azure-monitor-opentelemetry/CHANGELOG.md (hunk 3 above)
Comment (jeremydvoss) [reply sdk/monitor/azure-monitor-opentelemetry/CHANGELOG.md]: Done manually

**Review by lzchen — APPROVED** [review]

**Pull request merged**